tv_apps/smart-system/js/trusted_ui.js

```diff
@@ -110,15 +110,32 @@ var TrustedUIManager = {
       // class by the hide() method will trigger a 'transitionend'
       // event ultimately to be fired.
       this._hide();
-
-      window.focus();
+      // focus back to the top most window.
+      // XXX: we focus back to active app. But we should call the fallback
+      // algorithm to find the top-most overlay or app.
+      AppWindowManager.getActiveApp().getTopMostWindow().focus();
     } else {
       this._closeDialog(chromeEventId, origin);
     }
   },
 
-  isVisible: function trui_show() {
-    return this.screen.classList.contains('trustedui');
+  isVisible: function trui_isVisible() {
+    if (this.screen.classList.contains('trustedui')) {
+      var zIndex = window.getComputedStyle(this.overlay).zIndex;
+      return zIndex === 'auto' ? 0 : zIndex;
+    } else {
+      return false;
+    }
+  },
+
```

Comment:
Since 0 is consider not true, the equality condition have to change to this.isVisible() === false instead of !this.isVisible() like line 395, 422 and 445